### DISCUSSION TITLE
Default layouts

### DISCUSSION THREAD

**Discussion by mcDialogs** [whitecube/nova-flexible-content#397] [General]
Is it possible to force nova-flexible-content to start with some layouts already added to the field?

**Comment by voidgraphics** [comment]
You may be able to do that via the get method of a custom resolver class. [More info here](https://whitecube.github.io/nova-flexible-content/#/?id=custom-resolver-classes).

**Reply by johnpuddephatt** [reply]
I think (but might be wrong!) that @mcDialogs is talking about layouts appearing on the *field* – i.e. when on the create page for a resource within Nova.

I guess this would work something like the following:

```
Flexible::make("Content")
                    ->addLayout(\App\Nova\Flexible\Layouts\SimpleText::class)
                    ->addLayout(\App\Nova\Flexible\Layouts\Gallery::class)
                    ->addLayout(\App\Nova\Flexible\Layouts\Video::class)
                    ->defaultLayouts(['simple-text'])  // refers to name specified on Layout
```

I'm not sure if "defaultLayouts" is the right name. Perhaps "initialLayouts" because these are the initial layouts the field would start with? I don't know what is clearest.

With this option set, when visiting /nova/{resourceName}/new, there would be an empty text layout already within the field. This would be quite nice for something like a blog post, where you can start with a text block because that's what most blog posts are made up of. But it could also be good for something like a case study in a portfolio, where you could start with a series of blocks – text, image, quote etc to provide a good starting point to the user.

**Comment by martingetzel** [comment]
I know this discussion is a little older, but I managed to force a layout by default using a Custom Resolver as @voidgraphics said.
Here's what my Custom Resolver looks like:

```php
<?php

namespace App\Nova\Flexible\Resolvers;

use Illuminate\Support\Str;
use Whitecube\NovaFlexibleContent\Value\Resolver;
use Whitecube\NovaFlexibleContent\Value\ResolverInterface;

class LessonBodyResolver extends Resolver implements ResolverInterface
{
    /**
     * Resolve the Flexible field's content.
     *
     * @param  mixed  $resource
     * @param  string  $attribute
     * @param  \Whitecube\NovaFlexibleContent\Layouts\Collection  $layouts
     * @return  \Illuminate\Support\Collection<array-key, \Whitecube\NovaFlexibleContent\Layouts\Layout>
     */
    public function get($resource, $attribute, $layouts)
    {
        if (is_null($resource->$attribute)) {
            $value = json_decode('[{"layout":"textblock","key":"'. Str::random() .'","attributes":{}}]');
        } else {
            $value = $this->extractValueFromResource($resource, $attribute);
        }

        return collect($value)->map(function ($item) use ($layouts) {
            $layout = $layouts->find($item->layout);

            if (! $layout) {
                return null;
            }

            return $layout->duplicateAndHydrate($item->key, (array) $item->attributes);
        })->filter()->values();
    }
}
```
Of course, this could probably be improved, but at least I was able to set the `textblock` layout as the default one.
